RecoLocalCalo/HGCalRecProducers/interface/DumpClustersDetails.h

```diff
@@ -0,0 +1,188 @@
+#ifndef RecoLocalCalo_HGCalRecProducers_DumpClustersDetails_h
+#define RecoLocalCalo_HGCalRecProducers_DumpClustersDetails_h
+
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <algorithm>
+#include <ctime>
+#include <cstdlib>
+#include <unistd.h>  // For getpid
+#include <fmt/format.h>
+
+namespace hgcalUtils {
+
+  static bool sortByDetId(const std::pair<int, float>& pair1, const std::pair<int, float>& pair2) {
+    return pair1.first < pair2.first;
+  }
+
+  class DumpClustersDetails {
+  public:
+    DumpClustersDetails(){};
+
+    template <typename T>
+    void dumpInfos(const T& clusters, bool dumpCellsDetId = false) const {
+      // Get the process ID
+      pid_t pid = getpid();
+
+      // Create the filename using the PID
+      std::ostringstream filename;
+      // Seed the random number generator
+      srand(time(0));
+      // Generate a random number between 100 and 999
+      int random_number = 100 + rand() % 900;
+      filename << "CLUSTERS_" << pid << "_" << random_number << ".txt";
```

Comment:
I'm curious, why a random number in the file name ?

Reply:
Seeded and unseeded paths in the menu call the same producer with different labels. I wanted to have all the dumps around.

Reply:
I see.

Then, a suggestion could be
  - add a data member to `HGCalLayerClustersFromSoAProducer` 
    ```c++
    #if DEBUG_CLUSTERS_ALPAKA
    std::string moduleLabel_;
    #endif
    ```
  - initialise it in the constructor as 
    ```c++
    #if DEBUG_CLUSTERS_ALPAKA
    moduleLabel_{cfg.getParameter<std::string>("@module_label")},
    #endif
    ```
  - pass it as an argument to the `dumper`
  - use it here as part of the file name